### PR TITLE
Add recent event list to sidebar

### PR DESCRIPTION
**Pull request by RealOrangeOne**
Add functionality to add a list of events to the sidebar. 

image

### CODE REVIEW

File: layouts/partials/sidebar.html [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
+	{{ partial "widgets/recent-events.html" . }}
 	{{ partial "widgets/categories.html" . }}
 	{{ partial "widgets/social.html" . }}
 	{{ partial "widgets/taglist.html" . }}
 
 	{{- if and (not .Site.Params.widgets.search) (not .Site.Params.widgets.recent_articles) (not .Site.Params.widgets.categories) (not .Site.Params.widgets.social) (not .Site.Params.widgets.taglist) }}
 	<p class="sidebar__warning"><strong>WARNING:</strong><br>Please activate at least one sidebar widget.</p>
 	{{- end }}
-</aside>
+</aside>
```
Comment (PeterJCLaw) [inline layouts/partials/sidebar.html]: Trailing newline please (is your editor configured to remove them? If so, please change that -- lines aren't complete without their terminating newline character)

File: layouts/partials/sidebar.html [hunk 2]
```diff
@@ -1,11 +1,12 @@
 <aside class="sidebar" itemscope="itemscope" itemtype="http://schema.org/WPSideBar">
 	{{ partial "widgets/search.html" . }}
 	{{ partial "widgets/recent.html" . }}
+	{{ partial "widgets/recent-events.html" . }}
```
Comment (PeterJCLaw) [inline layouts/partials/sidebar.html]: Are they really going to be "recent" events?

File: layouts/partials/widgets/recent-events.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+            {{- $recent_articles_num := (.Site.Params.widgets.recent_articles_num | default 10) }}
+            {{- range first $recent_articles_num (where .Site.RegularPages "Section" "events") }}
+            <li class="widget__item">
+                <a class="widget__link" href="{{ .RelPermalink }}">{{ .Title }} - {{ .Date.Format "Jan 02, 2006" }}</a>
+            </li>
+            {{- end }}
+        </ul>
+    </div>
+</div>
+{{- end }}
```
Comment (PeterJCLaw) [inline layouts/partials/widgets/recent-events.html]: Trailing newline please

**Review by PeterJCLaw — CHANGES_REQUESTED** [review]
Nice! I'd been wondering about this too.

I'm not sure we should limit this to "recent" events though -- we probably want to also show upcoming ones.

File: layouts/partials/sidebar.html (hunk 1 above)
Comment (RealOrangeOne) [reply layouts/partials/sidebar.html]: This seems to be a recurring comment in my PRs, i'll sink some time into seeing what on earth is going on!

**Comment by RealOrangeOne** [comment]
What this actually does, is take the 10 newest events, it doesn't limit it to those which happen before now. Perhaps a better title is in order, although I'm not sure what to suggest?

**Comment by PeterJCLaw** [comment]
Perhaps just "events"?

**Comment by kierdavis** [comment]
"Current events" maybe?

**Review by PeterJCLaw — APPROVED** [review]

**Comment by PeterJCLaw** [comment]
For posterity: we went with "events" as the name because:
 * the list will likely include future and past events, with no limit on how long ago/ahead they are (at least in this implementation)
 * any filtering we do add will run at _build time_ rather than _view time_, meaning that adding filtering is complicated

Since showing all the events present in our site in one go is actually fine for us, "Events" captures that best.

We plan to look at filtering the list if/when that becomes an issue.

**Pull request merged**